File: app/src/main/java/begovac/rectangles/ui/home/HomeFragment.java
package begovac.rectangles.ui.home;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import javax.inject.Inject;

import begovac.rectangles.R;
import begovac.rectangles.model.Data;
import begovac.rectangles.repository.Repository;
import begovac.rectangles.ui.Navigator;
import begovac.rectangles.ui.choose.ChooseScreen;
import begovac.rectangles.ui.rectangles.RectanglesScreen;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;
import dagger.android.support.AndroidSupportInjection;

public class HomeFragment extends Fragment {

  @Inject Navigator navigator;
  @Inject Repository<Data> dataRepository;

  @BindView(R.id.btn_continue)
  Button btnContinue;

  Unbinder unbinder;

  public HomeFragment() {
    // Required empty public constructor
  }

  public static HomeFragment newInstance() {
    return new HomeFragment();
  }

  @Override
  public void onAttach(Context context) {
    AndroidSupportInjection.inject(this);
    super.onAttach(context);
  }

  @Override
  public View onCreateView(
      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.fragment_home, container, false);
    unbinder = ButterKnife.bind(this, view);
    return view;
  }

  @Override
  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    btnContinue.setEnabled(containsSomeData());
    btnContinue.setClickable(containsSomeData());
  }

  @Override
  public void onDestroyView() {
    super.onDestroyView();
    unbinder.unbind();
  }

  @OnClick(R.id.btn_start)
  public void onStartClicked() {
    navigator.navigate(ChooseScreen.create());
  }

  @OnClick(R.id.btn_continue)
  public void onContinueClicked() {
    navigator.navigate(RectanglesScreen.create());
  }

  private boolean containsSomeData() {
    return dataRepository.getFirst() != null;
  }
}

File: app/src/main/kotlin/begovac/rectangles/ui/home/HomeFragment.kt
package begovac.rectangles.ui.home

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import begovac.rectangles.R
import begovac.rectangles.model.Data
import begovac.rectangles.repository.Repository
import begovac.rectangles.ui.AppScreen
import begovac.rectangles.ui.Navigator
import begovac.rectangles.ui.navigate
import dagger.android.support.AndroidSupportInjection
import kotlinx.android.synthetic.main.fragment_home.*
import javax.inject.Inject

class HomeFragment : Fragment() {
  
  @Inject lateinit var navigator: Navigator
  @Inject lateinit var dataRepository: Repository<Data>
  
  companion object {
    fun newInstance(): HomeFragment {
      return HomeFragment()
    }
  }
  
  override fun onAttach(context: Context?) {
    AndroidSupportInjection.inject(this)
    super.onAttach(context)
  }
  
  override fun onCreateView(
      inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
      inflater.inflate(R.layout.fragment_home, container, false)
  
  override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    btn_continue.isEnabled = containsSomeData()
    btn_continue.isClickable = containsSomeData()
    
    btn_start.setOnClickListener { navigator.navigate(AppScreen.Choose) }
    btn_continue.setOnClickListener { navigator.navigate(AppScreen.Rectangles) }
  }
  
  private fun containsSomeData() = dataRepository.first != null
  
}
